ipatests/test_integration/test_trust.py

```diff
@@ -175,6 +220,114 @@ def test_upn_user_authentication_in_nonposix_trust(self):
         self.master.run_command(['kinit', '-C', '-E', self.upn_principal],
                                 stdin_text=self.upn_password)
 
+    def test_filter_root_user(self):
+        """filter_users option is applied if SSSD starts offline.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/3983
+
+        When domain resolution order changed to prefer the AD domain
+        calling id root then No LDAP lookups for 'root@ad.domain'.
+        """
+        domains = '{0}:{1}'.format(self.ad_domain, self.master.domain.name)
+        tasks.set_domain_resolution(self.master, order=domains)
+        log_file = '{0}/sssd_nss.log'.format(paths.VAR_LOG_SSSD_DIR)
+        logsize = tasks.get_logsize(self.master, log_file)
+        try:
+            # starting sssd in offline mode
+            self.master.run_command(['ipactl', 'stop'])
+            tasks.clear_sssd_cache(self.master)
+
+            self.master.run_command(['id', 'root'])
+            exp_txt1 = ('Looking up [root@{0}] in data provider'
+                        .format(self.ad_domain))
+            sssd_logs = self.master.get_file_contents(log_file)[logsize:]
+            assert exp_txt1.encode() not in sssd_logs
+        finally:
+            self.master.run_command(['ipactl', 'start'])
+            tasks.restore_files(self.master)
+            tasks.clear_sssd_cache(self.master)
+            tasks.set_domain_resolution(self.master, order='')
+
+    @pytest.mark.xfail(
+        osinfo.id == 'fedora' and osinfo.version_number <= (30,),
+        reason='sssd ticket 3978', strict=True)
+    def test_filter_users_offline(self):
+        """Negative cache should use values from 'filter_users' config option.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/3978
+
+        When there are users in filter_users for nss provider then no
+        look up should be in data provider and added in Negative cache
+        if sssd starts when ipa is offline and domain order to prefer AD.
+        """
+        domains = '{0}:{1}'.format(self.ad_domain, self.master.domain.name)
+        tasks.set_domain_resolution(self.master, order=domains)
```

Comment:
Are you sure we need to set domain resolution order in those tests? I do not see it being mentioned in related BZ. And we are using principal names here.